### ISSUE TITLE
[User Story] Mobile Version Of Contact Management

### ISSUE THREAD

**Issue by florian-papsdorf** [hpi-swt2/contacts-portal#97]
# Description
As a user I want to have a mobile version to manage my contacts because I usually access the platform with my smartphone.

# Conditions of satisfaction
- [ ] design follows wireframe below
- [ ] each contact has its own card
- [ ] on each card the full name (first name + last name) is shown
- [ ] the cards are sorted alphabetically
- [ ] each card shows which handles are saved for this contact
- [ ] the mobile version is automatically shown if the user accesses the portal with his/her smartphone

![Mobile Contacts](https://user-images.githubusercontent.com/49521681/103784220-f8a73e00-5039-11eb-95c9-f821dc3fc9de.png)

**Comment by LinqLover** [comment]
@florian-papsdorf Thanks for the detailed acceptance criteria! :-)

So unfortunately this one appears to have the following dependencies:
- [x] #69 for social handles [still WIP]
- [x] #26 for contact list [team2-scrumbledores-army, apparently only awaiting customer review and already merged in dev)) so this is still blocked ATM]
- [x] #91 for mobile layout? Maybe we can work around this, this would require additional investigation [two branches distant from the `dev` branch]

Aiui, the following aspects will *not* be part of this story. @florian-papsdorf Could you please create new issues for them if they are relevant for you?
- social media platform icons are not yet implemented anywhere -- for now, we could use a circle with the initial letter(s) of the platform name
- user photos are not yet implemented anywhere -- for now, we could use a circle with the initial letter(s) again (probably #68)
- navbar changes (position, look, and icons) are not part of this user story (probably #91)

---

Nevertheless, here are some implementational thoughts about this story:

- with #69 and #26, there are already working (but ugly) UIs for listing social handles/contacts
- we need some controller logic for retrieving the required data for the contact page
- this task will be very CSS-focused to create a good design (btw: bootstrap offers a card class we should probably use)
- for automatical selection of the right layout, there appears to be some kind of `variant` concept in #91

Tasks:

- [ ] write acceptance tests (some regex matches might be enough, so we can probably reuse other tests)
  - Should these tests also check the correct variant selection (mobile/not mobile)?
- [ ] write controller tests?
- [ ] write new controller logic
- [ ] write new views for cards and mobile contact list

Possible pitfalls:

- CSS layouts can be very tricky
- mobile layout selection (maybe depends on #91?)

**Comment by florian-papsdorf** [comment]
- There is a wireframe for #69 which includes icons for the different handles.
- Profile pictures should be implemented later, so at the moment, an empty circle with a default icon is enough.
- The issue describes mostly an UI task

**Comment by LinqLover** [comment]
Thanks for the reply!

> There is a wireframe for #69 which includes icons for the different handles.

No, sorry, #69 does not have an acceptance criterion for icons, so this has not been scheduled at the moment. Maybe open a new ticket for this?
